rebasehelper/tags.py

```diff
@@ -111,32 +137,35 @@ def sanitize(name):
                     m = tag_re.match(line)
                     if m:
                         span = cast(Tuple[int, int], tuple(x + start for x in m.span('value')))
-                        result.append(Tag(section, index, sanitize(m.group('name')), span, valid))
+                        result.append(Tag(section_index, section, index, sanitize(m.group('name')), span, valid))
         return cast(Tuple[Tag], tuple(result))
 
     @classmethod
-    def _filter(cls, tags: Tuple[Tag], section: Optional[str] = None, name: Optional[str] = None,
-                valid: Optional[bool] = True) -> Iterator[Tag]:
+    def _filter(cls, tags: Tuple[Tag], section_name: Optional[str] = None, name: Optional[str] = None,
+                valid: Optional[bool] = True, section_index: int = None) -> Iterator[Tag]:
         result = iter(tags)
-        if section is not None:
-            result = filter(lambda t: t.section == section.lower(), result)  # type: ignore
+        if section_name is not None:
+            result = filter(lambda t: t.section_name == section_name.lower(), result)  # type: ignore
         if name is not None:
             result = filter(lambda t: fnmatch.fnmatchcase(t.name, name.capitalize()), result)  # type: ignore
         if valid is not None:
             result = filter(lambda t: t.valid == valid, result)
+        if section_index is not None:
+            result = filter(lambda t: t.section_index == section_index, result)
         return result
 
-    def filter(self, section: Optional[str] = None, name: Optional[str] = None,
-               valid: Optional[bool] = True) -> Iterator[Tag]:
+    def filter(self, section_name: Optional[str] = None, name: Optional[str] = None,
+               valid: Optional[bool] = True, section_index: int = None) -> Iterator[Tag]:
         """Filters tags based on section, name or validity. Defaults to all valid tags in all sections.
 
         Args:
-            section: If specified, includes tags only from this section.
+            section_name: If specified, includes tags only from this section.
             name: If specified, includes tags matching this name. Wildcards are supported.
             valid: If specified, includes tags of this validity.
+            section_index: If specified, includes tags from the repeating section of this index.
```

Comment:
This description doesn't make sense to me. I would just use one line with both `section_index` and `section_name`.

Reply:
I don't think that 2 arguments on one line is a valid syntax. I fixed the description